src/sink.rs

```diff
@@ -32,9 +33,7 @@ impl<'a, H: ByteHash> Sink<'a, H> {
     pub(crate) fn fin(self) -> io::Result<H::Digest> {
         let Sink { bytes, .. } = self;
         let mut hasher = H::state();
-        hasher
-            .write_all(&bytes)
-            .expect("In memory write should always succeed");
+        hasher.write(&bytes);
```

Comment:
shouldn't we call `write_all`?

Reply:
This replaces io::Write with hasher::Hasher